norfair/lib_metrics.py

```diff
@@ -0,0 +1,336 @@
+
+        predictions_folder = os.path.join(save_path, "predictions")
+        if not os.path.exists(predictions_folder):
+            os.makedirs(predictions_folder)
+
+        out_file_name = os.path.join(predictions_folder, file_name + ".txt")
+        self.text_file = open(out_file_name, "w+")
+
+        self.frame_number = 1
+
+    def update_text_file(self, predictions):
+        write_predictions(
+            frame_number=self.frame_number, objects=predictions, out_file=self.text_file
+        )
+        self.frame_number += 1
+
+        if self.frame_number > self.length:
+            self.text_file.close()
+
+
+class DetFromFile:
+    """From txt files with the detections, get norfair detections"""
+
+    def __init__(self, input_path=None):
+        """get detecions matrix data with rows corresponding to:
+        frame, id, bb_left, bb_top, bb_right, bb_down, conf, x, y, z"""
+        if input_path is None:
+            raise ValueError(
+                "You must set 'input_path' argument when setting 'Det_from_file' class"
+            )
+        detections_path = os.path.join(input_path, "det/det.txt")
+        self.matrix_detections = process_text_file(path=detections_path)
+
+        seqinfo_path = os.path.join(input_path, "seqinfo.ini")
+        length = search_value_on_document(seqinfo_path, "seqLength")
+
+        self.ordered_by_frame = []
+
+        for frame_number in np.arange(1, length + 1):
+            dets_on_this_frame = self.get_dets_from_frame(frame_number)
+            self.ordered_by_frame.append(dets_on_this_frame)
+
+    def get_dets_from_frame(self, frame_number=None):
+        """ this function returns a list of norfair Detections class, corresponding to frame=frame_number """
+
+        if frame_number is None:
+            raise ValueError(
+                "You must set 'frame_number' argument when calling get_dets_from_frame()"
+            )
+
+        indexes = np.argwhere(self.matrix_detections[:, 0] == frame_number)
+        detections = []
+        if len(indexes) > 0:
+            actual_det = self.matrix_detections[indexes]
+            actual_det.shape = [actual_det.shape[0], actual_det.shape[2]]
+            for det in actual_det:
+                points = [[det[2], det[3]], [det[4], det[5]]]
+                points = np.array(points)
+                conf = det[6]
+                new_detection = Detection(
+                    points, np.array([1, 1])
+                )  # set to [1,1] or [conf,conf]
+                detections.append(new_detection)
+        self.actual_detections = detections
+        return detections
+
+
+class Accumulators:
+    def __init__(self):
+        self.matrixes_predictions = []
+        self.paths = []
+
+    def create_acc(self, input_path=None):
+        if not hasattr(self, 'matrixes_predictions'):
+            self.matrixes_predictions=[]
+        if not hasattr(self, 'paths'):
+            self.paths=[]
```

Comment:
Don't these get created in `__init__` and thus should always be an attribute of the object?